qiskit/providers/aer/pulse_new/models/transfer_functions.py

```diff
@@ -13,20 +13,102 @@
 # that they have been altered from the originals.
 
 from abc import ABC, abstractmethod
-from signals import Signal
+from typing import Callable, Union, List
+from .signals import Signal, ConstantSignal, PiecewiseConstant
+
+from numpy import convolve
+
 
 class BaseTransferFunction(ABC):
     """
     Base class for transforming signals.
     """
 
     @abstractmethod
-    def apply(self, signal: Signal) -> Signal:
+    def apply(self, signal: Union[Signal, List[Signal]]) -> Union[Signal, List[Signal]]:
         """
         Applies a transformation on a signal, such as a convolution,
         low pass filter, etc.
 
+        Args:
+            signal: A signal to which the transfer function will be applied.
+
         Returns:
             BaseSignal: The transformed signal.
         """
         raise NotImplementedError
+
+
+class Convolution(BaseTransferFunction):
+    """
+    Applies a convolution as a sum
+
+        (f*g)(n) = sum_k f(k)g(n-k)
+
+    The implementation is quadratic in the number of samples in the signal.
+    """
+
+    def __init__(self, func: Callable):
+        """
+        Args:
+            func: The convolution function specified in time.
+        """
+        self._func = func
+
+    def apply(self, signal: Union[Signal, List[Signal]]) -> Union[Signal, List[Signal]]:
+        """
+        Applies a transformation on a signal, such as a convolution,
+        low pass filter, etc. Once a convolution is applied the signal
+        can longer have a carrier as the carrier is part of the signal
+        value and gets convolved.
+
+        Args:
+            signal: A signal or list of signals to which the transfer function will be applied.
+
+        Returns:
+            signal: The transformed signal or list of signals.
+        """
+
+        if isinstance(signal, List):
+            convolved = []
+            for sig in signal:
+                convolved.append(self._convolve(sig))
+
+            return convolved
+        else:
+            return self._convolve(signal)
+
+    def _convolve(self, signal: Signal) -> Signal:
+        """
+        Helper function that applies the convolution to a single signal.
+
+        Args:
+            signal: The signal to convolve.
+
+        Returns:
+            signal: The transformed signal.
+        """
+        if isinstance(signal, ConstantSignal):
+            return signal
```

Comment:
Is this assuming the convolution function has a total integral/sum of `1`?

Reply:
Yes. But now that you point this out I realize that this is inconsistent with the rest of the code since I do not test that condition. I think for convolutions it makes sense to impose that the function we are convolving with is normalized to 1.
Scaling a signal should probably be a separate transfer function.